### ISSUE TITLE
Be able to hide the console startup messages

### ISSUE THREAD

**Issue by Yves57** [aspnet/Hosting#770]
When a Kestrel application has to be used "as is" by a customer, it could be fine to be able to not display the console initial messages.
For instance the current Hosting Environment or Content Root Path are not useful information. And moreover the messages are hard-coded in English.

Or course it is possible to write a custom Run() method, but it would be a shame.

I propose to add a new extension method `public static void Run(this IWebHost host, CancellationToken token, bool displayStartupMessages)`.

If you agree, I can do the job.

**Comment by muratg** [comment]
@Yves57 thanks for the recommendation. We're currently focused on RTM, and will have a look at this after we ship.

**Comment by davidfowl** [comment]
It might be cleaner to allow setting a `StreamWriter` as part of the `WebHostBuilder` configuration as a place where these messages would go by default. That way you can just generically redirect them across the board. 

We're using `Console.WriteLine` because logging may not have been configured yet.

**Comment by Tratcher** [comment]
> We're using Console.WriteLine because logging may not have been configured yet.

I don't think this is still true. Most/all of our Console.WriteLine statements are in Run, which is after all the initialization has happened. The only reason to continue using Console.WriteLine is that you may not have enabled a default logger.

**Comment by davidfowl** [comment]
https://github.com/aspnet/Hosting/search?utf8=%E2%9C%93&q=Console.WriteLine

**Comment by muratg** [comment]
The workaround is adding your own extension methods or writing a wrapper.

**Comment by cknaap** [comment]
Maybe you could add a parameter for a logger (ILogger kestrelLogger = null), or try to get an ILoggerFactory from the host.Services, and default to the Console if you don't get or can't create a logger.
There is a IWebHostBuilder.UseLoggerFactory, and I expect Kestrel to use it if I configure it.

**Comment by gingters** [comment]
Arg, I just created this issue to suggest to use an ILogger if available and fall back to Console only if there is no logger available: https://github.com/aspnet/Hosting/issues/979
